fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryManager.java

```diff
@@ -106,11 +106,16 @@ public CompletableFuture<Void> mergeIndex(@Nonnull LucenePartitioner partitioner
         final Subspace subspace = state.indexSubspace;
         final KeyExpression rootExpression = state.index.getRootExpression();
 
+        // This agilityContext will be used to determine/iterate grouping keys and partitions. The time gap between calls might
+        // be too long for a non-agile context.
         final AgilityContext agilityContext = getAgilityContext(true);
+
         if (! (rootExpression instanceof GroupingKeyExpression)) {
-            // TO DO: https://github.com/FoundationDB/fdb-record-layer/issues/2414
-            return mergeIndex(analyzerWrapper, TupleHelpers.EMPTY, partitioner, agilityContext);
+            // Here: empty grouping keys tuple
+            return mergeIndex(analyzerWrapper, TupleHelpers.EMPTY, partitioner, agilityContext)
+                    .whenComplete((vignore, eignore) -> agilityContext.flushAndClose());
```

Comment:
I don't think we want to call `flushAndClose()` here if there is an error, only if it is successful. If there is an error, I think we want `abortAndReset`

Reply:
✅